File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int maximum = 0;
        int left = 0;
        int right = heights.size() - 1;

        while (left < right) {
            int area = min(heights[left], heights[right]) * (right - left);
            maximum = max(area, maximum);
            if (heights[left] < heights[right]) {
                left++;
            } else right--;
        }
        return maximum;
    }
};

File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s_d = {}
        for char in s:
            if char not in s_d:
                s_d[char] = 1
            else:
                s_d[char] += 1
        t_d = {}
        for char in t:
            if char not in t_d:
                t_d[char] = 1
            else:
                t_d[char] += 1
        if s_d == t_d:
            return True
        else:
            return False

        